File: ios/Classes/StreamHandlers/LocalMediaStateStreamHandler.swift
//
//  LocalMediaStateStreamHandler.swift
//  sdk_live_flutter_plugin
//
//  Created by Ross on 01.11.2021.
//

import Foundation
import Flutter
import ScreenMeetSDK

class LocalMediaStateStreamHandler: NSObject, FlutterStreamHandler {
    private let pm = SwiftChannelParamsManager()

    var sink: FlutterEventSink?
       
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
       
    func sendMediaState(_ mediaState: SMParticipantMediaState, _ localRenderer: FlutterRTCVideoRenderer?) {
        guard let sink = sink else { return }
        sink(pm.mediaStateToDict(mediaState, localRenderer))
    }
       
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}

File: android/src/main/kotlin/com/screenmeet/sdk_live_flutter_plugin/handlers/LocalMediaStateStreamHandler.kt
package com.screenmeet.sdk_live_flutter_plugin.handlers

import com.screenmeet.sdk.ParticipantMediaState
import com.screenmeet.sdk_live_flutter_plugin.zipMediaState
import io.flutter.plugin.common.EventChannel

class LocalMediaStateStreamHandler: EventChannel.StreamHandler {

    private var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) { sink = events }
    override fun onCancel(arguments: Any?) { sink = null }

    fun sendMediaState(mediaState: ParticipantMediaState) {
        sink?.success(zipMediaState(mediaState))
    }
}